src/views/RejectionReasons.vue

```diff
@@ -114,36 +100,132 @@ export default defineComponent({
     IonTitle,
     IonToolbar,
   },
+  data() {
+    return {
+      queryString: '',
+      filteredReasons: []
+    }
+  },
+  computed: {
+    ...mapGetters({
+      rejectReasons: 'util/getRejectReasons',
+      rejectReasonEnumTypes: 'util/getRejectReasonEnumTypes',
+    })
+  },
+  async ionViewWillEnter() {
+    await this.store.dispatch('util/fetchRejectReasons')
+    await this.store.dispatch('util/fetchRejectReasonEnumTypes')
+    this.filteredReasons = this.rejectReasons ? JSON.parse(JSON.stringify(this.rejectReasons)) : []
+  },
   methods: {
     async openCreateRejectionReasonModal() {
       const createRejectionReasonModal = await modalController.create({
         component: CreateRejectionReasonModal
       });
 
+      createRejectionReasonModal.onDidDismiss().then(() => {
+        this.findFilteredReasons()
+      })
+
       createRejectionReasonModal.present()
     },
-    async openRejectionReasonActionsPopover(event: Event) {
+    async openRejectionReasonActionsPopover(event: Event, reason: any) {
       const popover = await popoverController.create({
         component: RejectReasonActionsPopver,
+        componentProps: { reason },
+        showBackdrop: false,
         event
       });
 
+      popover.onDidDismiss().then(() => {
+        this.findFilteredReasons()
+      })
+
       return popover.present();
     },
-    async openVarianceTypeActionsPopover(event: Event) {
+    async openVarianceTypeActionsPopover(event: Event, reason: any) {
       const varianceTypeActionsPopover = await popoverController.create({
         component: VarianceTypeActionsPopover,
+        componentProps: { reason },
+        showBackdrop: false,
         event
       });
 
+      varianceTypeActionsPopover.onDidDismiss().then(() => {
+        this.findFilteredReasons()
+      })
+
       return varianceTypeActionsPopover.present();
     },
+    findFilteredReasons() {
+      this.filteredReasons = this.rejectReasons.filter((reason: any) => reason.description.toLowerCase().includes(this.queryString.toLowerCase()))
+    },
+    async doReorder(event: CustomEvent) {
+      const previousSeq = JSON.parse(JSON.stringify(this.filteredReasons))
+
+      // returns the updated sequence after reordering
+      const updatedSeq = event.detail.complete(JSON.parse(JSON.stringify(this.filteredReasons)));
+
+      let diffSeq = this.findReasonsDiff(previousSeq, updatedSeq)
+
+      const updatedSeqenceNum = previousSeq.map((rejectionReason: any) => rejectionReason.sequenceNum)
+      Object.keys(diffSeq).map((key: any) => {
+        diffSeq[key].sequenceNum = updatedSeqenceNum[key]
+      })
+
+      diffSeq = Object.keys(diffSeq).map((key) => diffSeq[key])
+      this.filteredReasons = updatedSeq
+
+      if(diffSeq.length) {
```

Comment:
This will trigger multiple toast on UI when reordering multiple reasons without clicking save button.